lua/neo-tree/sources/filesystem/init.lua

```diff
@@ -369,6 +369,17 @@ M.setup = function(config, global_config)
     })
   end
 
+  if global_config.highlight_open_files then
+    for _, event in ipairs({ events.VIM_BUFFER_ADDED, events.VIM_BUFFER_DELETED }) do
+      manager.subscribe(M.name, {
+        event = event,
+        handler = function()
+          manager.refresh(M.name)
```

Comment:
This is the right place to collect all of the open buffers and add it to the state. Search for `modified_buffers_changed` for a good example of how to do this.

Reply:
Ah yes I misunderstood the logic behind state and didn't manage to pass it to the update function.